modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,931 @@
+
+  ADReal sumeps = 0.0;
+  ADReal sumepsx = 0.0;
+  ADReal sumepsy = 0.0;
+  ADReal sumepsz = 0.0;
+
+  ADReal sumL = 0.0;
+
+  ADReal sumepsxplus = 0.0;
+  ADReal sumepsyplus = 0.0;
+  ADReal sumepszplus = 0.0;
+
+  ADReal aniso = 0.0;
+  ADReal anisox = 0.0;
+  ADReal anisoy = 0.0;
+  ADReal anisoz = 0.0;
+
+  ADReal anisoxplus = 0.0;
+  ADReal anisoyplus = 0.0;
+  ADReal anisozplus = 0.0;
+
+  ADReal anisoL = 0.0;
+
+  ADReal Val = 0.0;
+  ADReal Val1 = 0.0;
+  ADReal sumval = 0.0;
+
+  ADReal f0saddle = 0.0;
+  ADReal g_gamma = 0.0;
+
+  ADReal eps_base = 0.0;
+  ADReal TotGaussValue = 0.0;
+  ADReal SumTotGaussValue = 0.0;
+  ADReal Totepsbar = 0.0;
+
+  ADReal eps_min = 0.0;
+  ADReal Totepsmin = 0.0;
+
+  f0saddle = 0.124961;
+
+  g_gamma = 0.471404;
+
+  // eps_base = ( (_sigmaBASE *_JtoeV * _length_scale * _length_scale) * _lgb *
+  // (std::sqrt(f0saddle)) ) / (g_gamma) ;
+  eps_base = (3.0 / 4.0) * (_sigmaBASE * _JtoeV * _length_scale * _length_scale) * _lgb;
+
+  //********************** sum (etai*etaj)^2 at each quadrature point **************************
+
+  for (unsigned int m = 0; m < _op_num - 1; ++m)
+  {
+    for (unsigned int n = m + 1; n < _op_num; ++n) // m<n
+    {
+      Val1 = (((*_vals[m])[_qp]) * ((*_vals[m])[_qp])) * (((*_vals[n])[_qp]) * ((*_vals[n])[_qp]));
+      sumval += Val1;
+    }
+  }
+
+  //********************** Main anisotopy calculation **************
+
+  for (unsigned int b = 0; b < _grain_num - 1; ++b)
+  {
+
+    for (unsigned int m = 0; m < _op_num - 1; ++m)
+    {
+
+      for (unsigned int a = b + 1; a < _grain_num; ++a)
+      {
+
+        for (unsigned int n = m + 1; n < _op_num; ++n)
+        {
+
+          if (b == op_to_grains[m])
+          {
+
+            if (a == op_to_grains[n])
+            {
+
+              aniso = eps_base;
+              anisox = 0.0;
+              anisoy = 0.0;
+              anisoz = 0.0;
+
+              anisoxplus = 0.0;
+              anisoyplus = 0.0;
+              anisozplus = 0.0;
+
+              anisoL = _L_BASE;
+
+              TotGaussValue = 0.0;
+
+              // calculate the continuous anisotropy part 1
+
+              Val = (((*_vals[m])[_qp]) * ((*_vals[m])[_qp])) *
+                    (((*_vals[n])[_qp]) * ((*_vals[n])[_qp]));
+
+              // vector of gaussian exponent
+
+              ADReal vsmalbax = (*_grad_vals[m])[_qp](0) - (*_grad_vals[n])[_qp](0);
+              ADReal vsmalbay = (*_grad_vals[m])[_qp](1) - (*_grad_vals[n])[_qp](1);
+              ADReal vsmalbaz = (*_grad_vals[m])[_qp](2) - (*_grad_vals[n])[_qp](2);
+
+              ADReal normvsmallbavalue =
+                  (vsmalbax * vsmalbax) + (vsmalbay * vsmalbay) + (vsmalbaz * vsmalbaz);
+
+              ADReal normvsmallba = std::pow(normvsmallbavalue, .5);
+
+              ADReal vsmallbax = vsmalbax / normvsmallba;
+              ADReal vsmallbay = vsmalbay / normvsmallba;
+              ADReal vsmallbaz = vsmalbaz / normvsmallba;
```

Comment:
```suggestion
              auto vsmallba  = vsmalba / vsmalba.norm();
```
then none of the stuff above would be necessary.
Also having two variables named `vsmalba` and `vsmallba` is confusing. We have no upper limit on identifier lengths. Please chose more expressive names.

Reply:
See updated files